src/mlpack/methods/ann/augmented/tasks/score_impl.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file score_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of scoring functions
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_SCORE_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_SCORE_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "score.hpp"
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace scorers /* Scoring utilities for augmented */ {
+
+double SequencePrecision(arma::field<arma::colvec> trueOutputs,
+                         arma::field<arma::colvec> predOutputs)
+{
+  double score = 0;
+  auto testSize = trueOutputs.n_elem;
```

Comment:
We should try to avoid ``auto``: I think if people go through the code using ``auto`` does not support the general code understanding, since you have to look further to figure out the return values. Also here is a quote from the armadillo page: 
> Can I use the C++11 auto keyword with Armadillo objects and/or expressions? 
> Use of C++11 auto is not recommended with Armadillo objects and expressions. Armadillo has a template meta-programming framework which creates lots of short lived temporaries that are not handled by auto.